addons/mail/models/mail_message.py

```diff
@@ -600,8 +617,9 @@ def _get_with_access(self, message_id, mode="read", **kwargs):
                 return message
 
         if message.model and message.res_id:
-            thread_mode = self.env[message.model]._get_mail_message_access([message.res_id], mode)
-            if self.env[message.model]._get_thread_with_access(message.res_id, mode=thread_mode, **kwargs):
+            thread_su = self.env[message.model].browse(message.res_id).sudo()
+            access_mode = thread_su._mail_get_operation_for_mail_message_operation(mode)[thread_su]
+            if access_mode and self.env[message.model]._get_thread_with_access(message.res_id, mode=access_mode, **kwargs):
```

Comment:
This is basically a duplicate of `_get_thread_with_access_for_post`, can't it be common code instead?

Reply:
The portal override must be applied here too, `kwargs` are forwarded here for this purpose. It's not controller specific, just validating access when token stuff are provided.
But the override applied here already, because the override is done on `_get_thread_with_access` which is models.

But ok either way, as indeed we are just duplicating 2 lines.